### ISSUE TITLE
Deprecate `t.error()`

### ISSUE THREAD

**Issue by sindresorhus** [avajs/ava#995]
From: https://github.com/avajs/eslint-plugin-ava/pull/136#issuecomment-238988505

We changed the name to `t.ifError` at some point and just aliased `t.error` to it, but never officially deprecated `t.error`.

Todo:

- [ ] Remove from here https://github.com/avajs/ava/blob/8816faf6bee675213c5bfe32f628a0a683708484/lib/assert.js#L150
- [ ] Add as deprecated here https://github.com/avajs/ava/blob/8816faf6bee675213c5bfe32f628a0a683708484/lib/assert.js#L162
- [ ] Create codemod for it https://github.com/avajs/ava-codemods

If you're looking to contribute, feel free to only do the first two items. The third one is harder.

**Comment by platy11** [comment]
@sindresorhus is this right: platy11/ava@83899e77b89264ba134142106c3ee55ef5662c2d

**Comment by sindresorhus** [comment]
@platy11 Yup, looks good.

**Comment by platy11** [comment]
I opened #999

**Comment by platy11** [comment]
@sindresorhus Looking at https://github.com/avajs/ava-codemods/blob/master/lib/ok-to-truthy.js, would you only need to change the references: `ok` to `error` etc.? (And add a test)

**Comment by sindresorhus** [comment]
Yes, and add it to https://github.com/avajs/ava-codemods/blob/master/codemods.json (add a 0.17.0 version)

**Comment by platy11** [comment]
@sindresorhus Thanks, doing that now

**Comment by platy11** [comment]
@sindresorhus is this correct for `codemods.json`?
https://github.com/platy11/ava-codemods/blob/master/codemods.json

**Comment by sindresorhus** [comment]
Yes, looks good. Also needs to be added to the readme: https://github.com/avajs/ava-codemods#supported-codemods

**Comment by platy11** [comment]
@sindresorhus What should I put under `Available upgrades`? 0.14.x to 0.17.x?
```
$ ava-codemods --help

  Usage
    $ ava-codemods [<file|glob> ...]

  Options
    --force, -f    Bypass safety checks and forcibly run codemods

  Available upgrades
    - 0.13.x → 0.14.x
```

**Comment by platy11** [comment]
@sindresorhus I did `0.16.x → 0.17.x` and opened PR avajs/ava-codemods#27. Feel free to share any things I've missed or improvements that could be made there.
